backend/GymBeam/CommandHandlers/ReservationCommandHandler.cs

```diff
@@ -52,5 +53,25 @@ public async Task<Unit> Handle(CreateReservationCommand request, CancellationTok
 
             return Unit.Value;
         }
+
+        public async Task<Unit> Handle(DeleteReservationCommand request, CancellationToken cancellationToken)
+        {
+            var reservation = _repository.GetReservation(request.ReservationId)
+                ?? throw new ReservationNotFoundException(request.ReservationId);
+
+            var loggedUserId = JwtHelper.GetUserIdFromCookies(_httpContextAccessor)
+                ?? throw new InvalidCookieException(Cookies.UserId);
+
+            var userRole = JwtHelper.GetRoleClaimFromCookie(_httpContextAccessor)
+                ?? throw new AuthenticationFailureException(Resource.ExceptionUserRoleNotFound);
+
+            if (userRole != Role.Admin.ToString() && reservation.User.Id != loggedUserId)
```

Comment:
I am not sure if NullPointerException may occur here (In case reservation.User == null). If the User field in the Reservation class is a required field, it should actually be handled at the database level and the null value should not appear, but maybe I should still add a check like: if (reservations.User == null) throw new InvalidOperationException? Is it necessary in this case?

Reply:
In my opinion its not required because we mark it in Domain with [Required] annotation